frontend/src/app/paypal/page.tsx

```diff
@@ -0,0 +1,213 @@
+          Make a Difference with Your Donation
+        </h2>
+        <p className="text-center text-gray-600 mb-6">
+          Your contribution brings positive change to many lives. Join us in
+          making a difference.
+        </p>
+
+        {errorMessage && (
+          <p className="text-red-500 text-sm mb-4">{errorMessage}</p>
+        )}
+
+        <form onSubmit={validateAndSubmit}>
+          <div className="mb-4">
+            <label
+              htmlFor="email"
+              className="block text-gray-700 font-medium mb-1"
+            >
+              Email <span className="text-red-500">*</span>
+            </label>
+            <input
+              type="email"
+              id="email"
+              value={email}
+              onChange={(e) => setEmail(e.target.value)}
+              placeholder="mustermann@beispiel.com"
+              required
+              className="w-full border border-gray-300 rounded-md p-2"
+            />
+          </div>
+
+          <div className="mb-4">
+            <label
+              htmlFor="amount"
+              className="block text-gray-700 font-medium mb-1"
+            >
+              Donation Amount <span className="text-red-500">*</span>
+            </label>
+            <div className="relative">
+              <span className="absolute left-3 top-2.5 text-gray-500">€</span>
+              <input
+                type="text"
+                id="amount"
+                value={amount}
+                onChange={(e) => formatAmount(e.target.value)}
+                placeholder="0.00"
+                required
+                className="w-full pl-8 border border-gray-300 rounded-md p-2"
+              />
+              <datalist id="amounts">
+                <option value="5" />
+                <option value="25" />
+                <option value="50" />
+                <option value="100" />
+              </datalist>
+            </div>
+          </div>
+
+          <div className="mb-4">
+            <label
+              htmlFor="purpose"
+              className="block text-gray-700 font-medium mb-1"
+            >
+              Donation Purpose
+            </label>
+            <select
+              id="purpose"
+              value={purpose}
+              onChange={(e) => setPurpose(e.target.value)}
+              className="w-full border border-gray-300 rounded-md p-2"
+            >
+              {/* Default option that can be selected */}
+              <option value="empty">
+                Choose a purpose (or clear selection)
+              </option>
+              {purposeOptions.length > 0 ? (
+                purposeOptions.map((option) => (
+                  <option key={option} value={option}>
+                    {option}
+                  </option>
+                ))
+              ) : (
+                <option value="">Loading options...</option>
+              )}
+            </select>
+          </div>
+
+          <button
```

Comment:
Could you use the shadcn/ui components to have a cleaner and easier changeable user interface?